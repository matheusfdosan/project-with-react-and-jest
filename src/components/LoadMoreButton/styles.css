#load-more {
  width: 100%;
  margin-top: 4rem;
  border: .1rem solid #37373788;
  border-radius: .3rem;
  padding: 1rem 0;
  font-family: "Playir Display", serif;
  font-size: 2rem;
  cursor: pointer;
  background-color: #fff;
  text-transform: uppercase;
  transition: background-color 200ms ease, letter-spacing 200ms ease;
}

#load-more:hover {
  background-color: #37373728;
  letter-spacing: .2rem;
}