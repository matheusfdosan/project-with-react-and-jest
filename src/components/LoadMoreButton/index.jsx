import React from "react"
import "./styles.css"

export default function LoadMoreButton({ onClickLoadMore }) {
  return <button onClick={onClickLoadMore} id="load-more">Load More</button>
}
