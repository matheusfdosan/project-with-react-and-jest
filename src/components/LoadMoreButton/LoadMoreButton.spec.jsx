import React from "react"
import "@testing-library/jest-dom"
import userEvent from "@testing-library/user-event"
import { fireEvent, render, screen } from "@testing-library/react"
import LoadMoreButton from "."

describe("<LoadMoreButton />", () => {
  it('should render text "Load More"', () => {
    render(<LoadMoreButton />)
    expect.assertions(2) // Nesse teste deve ter 2 expects (mais comum em testes assíncronos)

    // Normalmente, não vemos ninguém criando variáveis
    expect(
      screen.getByRole("button", { name: "Load More" })
    ).toBeInTheDocument()
    expect(screen.getByRole("button", { name: "Load More" })).toHaveAttribute(
      "id",
      "load-more"
    )
  })

  it("should call function on click", () => {
    const fn = jest.fn() // Mock de função
    render(<LoadMoreButton onClickLoadMore={fn} />)

    expect.assertions(1)

    const button = screen.getByRole("button", { name: "Load More" })

    fireEvent.click(button) // simulando o click no botão
    //await userEvent.click(button) // simulando o click no botão, deve ser assíncrono

    expect(fn).toHaveBeenCalledTimes(1)
  })

  it('should match snapshot', () => {
    const fn = jest.fn()
    const {container} = render(<LoadMoreButton onClickLoadMore={fn} />)
    expect(container.firstChild).toMatchSnapshot()
  });
})
