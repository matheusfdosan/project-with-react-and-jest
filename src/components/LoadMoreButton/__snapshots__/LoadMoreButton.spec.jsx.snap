// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMoreButton /> should match snapshot 1`] = `
<button
  id="load-more"
>
  Load More
</button>
`;
